File: Array3.java
import java.lang.*;
import java.util.*;

class Array3
{
    public static void main(String arg[])
    {
        int i = 0, j = 0;
        //arr is 2 dimensional array which contains 2 one dimensional array
        //each one dimensional array contains 3 elements and each element
        //is of type integer
        int arr[][] = new int[2][3];

        arr[0][0] = 11;
        arr[0][1] = 21;
        arr[0][2] = 51;
        arr[1][0] = 101;
        arr[1][1] = 111;
        arr[1][2] = 121;
        System.out.println("Size of array is:  "+arr.length);
        //arr.length 2
        //      1       2          3
        for(i = 0; i<arr.length; i++)   //Outer
        {   //  1           2           3
            for(j = 0; j<arr[i].length; j++) //Inner-|
            {//                                      |
                System.out.println(arr[i][j]);  //4  |-----//4 Outer cha
            }//--------------------------------------|
        }
    }
}

File: Array3.c
#include<stdio.h>

int main();
{
	//Arr is one dimentional array which contains 4 elements 
	//each element is of type integer

	int Arr[4];
	
	//Brr is two dimentional array which contains 3 one dimentional array
	// each one dimentional array contains 4 elements 
	//and each elements is of type integer
	
	int Brr[3][4];
	
	//Drr is two dimentional array which contains 5 one dimentional array
	// each one dimentional array contains 3 elements 
	//and each elements is of type double
	
	double Drr[5][3];
	
	// p is a pointer which holds address of integer,currently it hold address of no,
	//where no is a variable of type integer initialized with the value 11
	
	int no = 11;
	int*p = &no;	
	
	// no			11
	// &no			100
	// p			100
	// &p			200
	//*p			11
	// sizeof(no)	4
	// sizeof(p)	8
	// sizeof(*p)	4	
	
	return 0;
	
}